pkgs/by-name/cy/cyrus-imapd/package.nix

```
@@ -0,0 +1,191 @@
+{
+  autoreconfHook,
+  bison,
+  brotli,
+  coreutils,
+  cunit,
+  cyrus-imapd,
+  cyrus_sasl,
+  fetchFromGitHub,
+  fig2dev,
+  flex,
+  icu,
+  jansson,
+  lib,
+  libbsd,
+  libcap,
+  libchardet,
+  libical,
+  libmysqlclient,
+  libsrs2,
+  libuuid,
+  libxml2,
+  makeWrapper,
+  nghttp2,
+  openssl,
+  pcre2,
+  perl,
+  pkg-config,
+  postgresql,
+  rsync,
+  shapelib,
+  sqlite,
+  stdenv,
+  testers,
+  unixtools,
+  valgrind,
+  withCalalarmd ? true,
+  withHttp ? true,
+  withJMAP ? true,
+  withMySQL ? false,
+  withNNTP ? false,
+  withPgSQL ? false,
+  withSQLite ? true,
+  withXapian ? true,
+  wslay,
+  xapian,
+  zlib,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cyrus-imapd";
+  version = "3.8.2";
+
+  src = fetchFromGitHub {
+    owner = "cyrusimap";
+    repo = "cyrus-imapd";
+    rev = "cyrus-imapd-${finalAttrs.version}";
+    hash = "sha256-5I4EaLnxu2M1a79XmxUh1PXq703EfI73/9aohKg7jyc=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    pkg-config
+    autoreconfHook
+  ];
+  buildInputs =
+    [
+      unixtools.xxd
+      pcre2
+      flex
+      valgrind
+      fig2dev
+      perl
+      cyrus_sasl.dev
+      icu
+      jansson
+      libbsd
+      libuuid
+      openssl
+      zlib
+      bison
+      libsrs2
+    ]
+    ++ lib.optional stdenv.isLinux libcap
+    ++ lib.optionals (withHttp || withCalalarmd || withJMAP) [
+      brotli.dev
+      libical.dev
+      libxml2.dev
+      nghttp2.dev
+      shapelib
+    ]
+    ++ lib.optionals withJMAP [
+      libchardet
+      wslay
+    ]
+    ++ lib.optional withXapian [
```

Here's you're wrapping twice the items. That's one of the reason I hate using `lib.optional`.

```suggestion
    ++ lib.optionals withXapian [
```